applications/RANSApplication/custom_processes/rans_nut_nodal_update_process.cpp

```diff
@@ -0,0 +1,134 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <cmath>
+#include <limits>
+
+// External includes
+
+// Project includes
+#include "includes/cfd_variables.h"
+#include "includes/define.h"
+#include "utilities/parallel_utilities.h"
+
+// Application includes
+#include "rans_application_variables.h"
+
+// Include base h
+#include "rans_nut_nodal_update_process.h"
+
+namespace Kratos
+{
+RansNutNodalUpdateProcess::RansNutNodalUpdateProcess(
+    Model& rModel,
+    Parameters rParameters)
+: mrModel(rModel)
+{
+    KRATOS_TRY
+
+    rParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+
+    mEchoLevel = rParameters["echo_level"].GetInt();
+    mModelPartName = rParameters["model_part_name"].GetString();
+
+    KRATOS_CATCH("");
+}
+
+RansNutNodalUpdateProcess::RansNutNodalUpdateProcess(
+    Model& rModel,
+    const std::string& rModelPartName,
+    const int EchoLevel)
+: mrModel(rModel),
+  mModelPartName(rModelPartName),
+  mEchoLevel(EchoLevel)
+{
+}
+
+int RansNutNodalUpdateProcess::Check()
+{
+    KRATOS_TRY
+
+    const auto& r_model_part = mrModel.GetModelPart(mModelPartName);
+
+    KRATOS_ERROR_IF(!r_model_part.HasNodalSolutionStepVariable(VISCOSITY))
+        << "VISCOSITY is not found in nodal solution step variables list of "
+        << mModelPartName << ".";
+
+    KRATOS_ERROR_IF(!r_model_part.HasNodalSolutionStepVariable(TURBULENT_VISCOSITY))
+        << "TURBULENT_VISCOSITY is not found in nodal solution step variables list of "
+        << mModelPartName << ".";
+
+    return 0;
+
+    KRATOS_CATCH("");
+}
+
+void RansNutNodalUpdateProcess::ExecuteInitializeSolutionStep()
+{
+    KRATOS_TRY
+
+    if (!mIsInitialized) {
+        this->ExecuteAfterCouplingSolveStep();
+        mIsInitialized = true;
+    }
+
+    KRATOS_CATCH("");
+}
+
+void RansNutNodalUpdateProcess::ExecuteAfterCouplingSolveStep()
+{
+    KRATOS_TRY
+
+    auto& r_model_part = mrModel.GetModelPart(mModelPartName);
+    auto& r_nodes = r_model_part.Nodes();
+
+    const auto& r_properties = r_model_part.Elements().front().GetProperties();
+    const double rho = r_properties.GetValue(DENSITY);
+    const double nu = r_properties.GetValue(DYNAMIC_VISCOSITY) / rho;
+
+    block_for_each(r_nodes, [&](ModelPart::NodeType& rNode) {
+        rNode.FastGetSolutionStepValue(VISCOSITY) =
```

Comment:
We should stop using `VISCOSITY` as it can be easily misunderstood. I know that this is out of the scope of this PR but would be really good to change it to the `KINEMATIC_VISCOSITY` variable in `cfd_variables.h` (IMO `VISCOSITY` should be moved to `deprecated_variables.h` too).

Reply:
I agree... Once we atleast update `FractionalStep` and remove old `VMS` formulations, then we can totally get rid of the `VISCOSITY` variable from Fluid app for good ;)